include/graaflib/graph.tpp

```diff
@@ -201,4 +215,29 @@ void graph<VERTEX_T, EDGE_T, GRAPH_TYPE_V>::remove_edge(
   std::abort();
 }
 
+template <typename VERTEX_T, typename EDGE_T>
+void get_transposed_graph(
+    const directed_graph<VERTEX_T, EDGE_T>& graph,
+    directed_graph<VERTEX_T, EDGE_T>& out_transposed_graph) {
+  for (const auto [edge_vertices, edge] : graph.get_edges()) {
+    const auto [vertex_id_lhs, vertex_id_rhs] = edge_vertices;
+
+    auto vertex_value_lhs = graph.get_vertex(vertex_id_lhs);
+    auto vertex_value_rhs = graph.get_vertex(vertex_id_rhs);
+    auto edge_type = edge;
+
+    if (!out_transposed_graph.has_vertex(vertex_id_lhs)) {
+      auto vertex_id =
+          out_transposed_graph.add_vertex(vertex_value_lhs, vertex_id_lhs);
+    }
+
+    if (!out_transposed_graph.has_vertex(vertex_id_rhs)) {
+      auto vertex_id =
+          out_transposed_graph.add_vertex(vertex_value_rhs, vertex_id_rhs);
+    }
```

Comment:
Since the vertex value could be expensive to copy, we could use moves here 😉 

```suggestion
    if (!out_transposed_graph.has_vertex(vertex_id_lhs)) {
      auto vertex_id =
          out_transposed_graph.add_vertex(std::move(vertex_value_lhs), vertex_id_lhs);
    }

    if (!out_transposed_graph.has_vertex(vertex_id_rhs)) {
      auto vertex_id =
          out_transposed_graph.add_vertex(std::move(vertex_value_rhs), vertex_id_rhs);
    }
```

Reply:
Awesome, love this! Will add.